internal/pkg/git/git.go

```diff
@@ -201,7 +201,7 @@ func (r *Repository) Pull(ctx context.Context) error {
 }
 
 func (r *Repository) Free() <-chan struct{} {
-	done := make(chan struct{})
+	done := make(chan struct{}, 1)
```

Comment:
Nasiel som goroutines leak (neviem teda ci vsetky, ale tieto 3 isto).

Ak si pamatas, tak tu pouzivam `chanell` na signalizaciu toho, ze operacia dobehla.
Da sa to pozom pouzit synchronne, ak z `ch` citam, a teda blokujem kod, kym sa tam nezapise.
Alebo asynchronne, ked `ch` ignorujem.

Problem je v tom, ze aj zapis do `ch` moze byt blokujuci, ak je `ch` plny.
V tomto pripade vytvaram `ch` s `buffer=0`, tj. `make(chan struct{})` zodpoveda `make(chan struct{}, 0)`.

Zapis do `ch` mi teda zablokuje goroutine nizsie a ta neskonci, 
pretoze v produkcnom kode z `ch` necitam, iba v testoch.

Upravil som to tak, aby `buffer=1`, a teda zapis bude neblokujuci (z kodu vyplyva, ze tam nikdy nezapisujeme viac ako 1 hodnotu).

To ze `ch` nezavriem alebo z neho necitam nie je problem, ... o `ch` sa postara GC.
https://stackoverflow.com/questions/8593645/is-it-ok-to-leave-a-channel-open


-----

Daju sa na to napisat testy pomocou `goleak` libky.
Ale je tam par problemov:
- Nefunguje to s `t.Parallel()`, kedze paralelne testy bezia vramci jedneho procesu. Kontrola `goroutines` na konci testu teda zachyti ostatne goroutines.
- `etcd`, `http` a podobne kniznice si vytvaraju nejake goroutines, ktore zostavaju aj po teste. Aj ked zavolam `cancelFn()` pre dany `context`, tak to chvilu trva, kym sa dane goroutines ukoncia, a nie je ako na nich pockat a az tak ukoncit test. 
- Teda test vzhladom na narocnost/cas nebude. Mal by stacit DD.